### DISCUSSION TITLE
react-pdf/pdfjs - how to migrate cy.route to cy.interception

### DISCUSSION THREAD

**Discussion by joaosaro** [cypress-io/cypress#28031] [Questions and Help]
Hi!
I'm trying to upgrade Cypress from v6v to v12/13 but i need to change from `cy.route` to `cy.interception`. So far, I'm having a issue with a pdf file fixture. It works well in the app, but not in Cypress when change to interception method.

Previous
```
cy.route({
	method: 'GET',
	url: '{endpoint}',
	status: 200,
	response: 'fixture:file.pdf,binary',
```

I tried first to change to this and it returns an error "Warning: Invalid stream: "FormatError: Bad FCHECK in flate stream: 120, 239".

```
cy.interception({
	method: 'GET',
	url: '{endpoint}'
  }, {
       status: 200,
       fixture: 'file.pdf'
})
```

I also tried this:

```
cy.fixture('file.pdf', "binary").then((file) => {
	return cy.intercept(
	      {
		   method: 'GET',
		   url: '{endpoint}',
	       },
		{
			body: Cypress.Blob.binaryStringToBlob(file, 'application/pdf'),
			headers: {
				'Content-Type': 'application/pdf',
		        }
                 )}
      )}
```

The response is similar but the data buffer is almost the double. I looked into the documentation and other sources but I'm not getting successful answers. Anyone had this issue or know a possible cause?

**Comment by portyu9** [comment]
Hi @joaosaro 

For a PDF, preserve the fixture as **binary bytes**. With `cy.intercept()`, Cypress supports a `,null` encoding suffix specifically for loading a fixture as a `Buffer`.

Use:

```js
cy.intercept('GET', endpoint, {
  statusCode: 200,
  fixture: 'file.pdf,null',
  headers: {
    'content-type': 'application/pdf',
  },
})
```

There are two API-name differences from the code in the question:

- the command is `cy.intercept()`, not `cy.interception()`
- a `StaticResponse` uses `statusCode`, not `status`

The `,null` is the important part for the PDF corruption. Without it, the fixture can be decoded/transformed as text instead of preserving the original binary bytes, which is consistent with PDF.js then reporting a malformed compressed stream.

You should not need to turn the file into a `Blob` yourself for this use case.

Please mark this answer as accepted if it helped, thank you.
